### PR TITLE
feat(eslint-plugin): add new rule no-require-imports

### PR DESCRIPTION
**Pull request by lukyth**
Adds the equivalent of TSLint's [`no-require-imports`](https://palantir.github.io/tslint/rules/no-require-imports/) rule.

### CODE REVIEW

File: packages/eslint-plugin/docs/rules/no-require-imports.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+Examples of **correct** code for this rule:
+
+```ts
+import { l } from 'lib';
+var lib3 = load('not_an_import');
+var lib4 = lib2.subImport;
+var lib7 = 700;
+import lib9 = lib2.anotherSubImport;
```
Comment (bradzacher) [inline packages/eslint-plugin/docs/rules/no-require-imports.md]: just for completeness, could you also add `import lib10 from 'lib'`.

File: packages/eslint-plugin/lib/rules/no-require-imports.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      CallExpression(node) {
+        if (node.callee.name === 'require') {
+          context.report({
+            node,
+            message: 'A `require()` style import is forbidden.'
```
Comment (bradzacher) [inline packages/eslint-plugin/lib/rules/no-require-imports.js]: please use `messageId`: https://eslint.org/docs/developer-guide/working-with-rules#messageids

**Comment by j-f1** [comment]
Just thinking — would this be better suited as a rule in `eslint-plugin-import`?

**Comment by lukyth** [comment]
Not sure what's best here. I can try moving this over to `eslint-plugin-import` if you think it's more appropriate.

**Comment by sindresorhus** [comment]
I think the rule should stay here. `eslint-plugin-import` is not very actively maintained and it also doesn't currently have any rules that target TypeScript.

**Comment by JamesHenry** [comment]
Apologies for the conflicts, we decided to remove the counts from the ROADMAP here: https://github.com/typescript-eslint/typescript-eslint/pull/225

They are too awkward to maintain manually, so this is the last time you will have to deal with syncing that up!

**Pull request merged**